library/network_connections.py

```diff
@@ -1010,23 +1007,71 @@ def delete_cb(connection, result, cb_args):
 
         # workaround libnm oddity. The connection may not yet be gone if the
         # connection was active and is deactivating. Wait.
-        wait_count = 0
+        c_uuid = connection.get_uuid()
+        gone = self.wait_till_connection_is_gone(c_uuid)
+        if not gone:
+            raise MyError(
+                "connection %s was supposedly deleted successfully, but it's still here"
+                % (c_uuid)
+            )
+
+    def volatilize_connection(self, connection, timeout=10):
+        update2_cb = Util.create_callback("update2_finish")
+
+        cancellable = Util.create_cancellable()
+        cb_args = {}
+
+        connection.update2(
+            None,  # settings
+            Util.NM().SettingsUpdate2Flags.IN_MEMORY_ONLY
+            | Util.NM().SettingsUpdate2Flags.VOLATILE,  # flags
+            None,  # args
+            cancellable,
+            update2_cb,
+            cb_args,
+        )
+
+        if not Util.GMainLoop_run(timeout):
+            cancellable.cancel()
+            raise MyError("failure to volatilize connection: %s" % ("timeout"))
+
+        # Do not check of success if ther connection does not exist anymore
```

Comment:
maybe here issue a `Util.GMainLoop_iterate_all()`? `GMainLoop_run()` will quit once the callback returns, but there might be still other D-Bus messages pending. I think we should process them all before checking `if connection not in self.connection_list()`.